src/data-types/mailstream_cfstream.c

```diff
@@ -995,7 +996,8 @@ int mailstream_cfstream_set_ssl_enabled(mailstream * s, int ssl_enabled)
         count = SecTrustGetCertificateCount(secTrust);
         CFRelease(secTrust);
     }
-    else {
+    // This is work around for Mac OS X 10.6.
+    if(!secTrust || count == 0)  {
```

Comment:
If think testing count == 0 should be enough. Could you merge it with the test below?